### ISSUE TITLE
Bin-crate build fail with new LLD linker in windows (linker-arguments: invalid argument)

### ISSUE THREAD

**Issue by edwin0cheng** [rust-lang/rust#48948]
We just try to update to the new nightly compiler to build my project. 
```
rustc --version -> 1.26.0-nightly (2789b067d 2018-03-06)
```

but it fail to build in my Windows machine (it is working in nightly-2018-02-28), and the error message is following:

```
cargo build --target=wasm32-unknown-unknown
[...]
  = note: C:\Users\jice\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\x86_64-pc-windows-msvc\bin\lld.exe: error: no output file specified
          C:\Users\jice\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\x86_64-pc-windows-msvc\bin\lld.exe: error: cannot open @C:\Users\jice\AppData\Local\Temp\rustc.EgJv1gS5jdF9\linker-arguments:
invalid argument
          C:\Users\jice\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\x86_64-pc-windows-msvc\bin\lld.exe: error: no input files
```

And then i tested in my mac machine, it works. It seem to be Windows only problem ?

**Comment by edwin0cheng** [comment]
Original Post. https://github.com/koute/cargo-web/issues/80

**Comment by alexcrichton** [comment]
Thanks for the report! Turns out while linkers like `link.exe` accept arguments of the form `@file` LLD does not! We should probably disable [this logic](https://github.com/rust-lang/rust/blob/222b0eb583a64299f43d6640617e22f19973ddab/src/librustc_trans/back/link.rs#L835-L916) and [this conditional](https://github.com/rust-lang/rust/blob/222b0eb583a64299f43d6640617e22f19973ddab/src/librustc_trans/back/link.rs#L827) on Windows

**Comment by edwin0cheng** [comment]
Just checked, it fixed in rustc 1.26.0-nightly (392645394 2018-03-15), i will close this issue now, thank you so much.
